src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -0,0 +1,516 @@
+            TimeTriggerLimit::Month(num) => {
+                let num_years_start = time_start.year() as u64;
+                let num_months_start = num_years_start * 12 + time_start.month() as u64;
+                let num_years_now = time_now.year() as u64;
+                let num_months_now = num_years_now * 12 + time_now.month() as u64;
+
+                num_months_now - num_months_start >= num
+            }
+            TimeTriggerLimit::Year(num) => {
+                let num_years_start = time_start.year() as u64;
+                let num_years_now = time_now.year() as u64;
+
+                num_years_now - num_years_start >= num
+            }
+        };
+        if is_triger {
+            let tmp = TimeTrigger::new(self.limit);
+            let time_new = tmp.time_start.read().unwrap();
+            *time_start = *time_new;
+        }
+        Ok(is_triger)
+    }
+
+    fn is_pre_process(&self) -> bool {
+        true
+    }
+}
+
+/// A deserializer for the `TimeTrigger`.
+///
+/// # Configuration
+///
+/// ```yaml
+/// kind: time
+///
+/// # The time limit in seconds. The following units are supported (case insensitive):
+/// # "second(s)", "minute(s)", "hour(s)", "day(s)", "week(s)", "month(s)", "year(s)". The unit defaults to
+/// # second if not specified.
+/// limit: 7 day
+/// ```
+#[cfg(feature = "config_parsing")]
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default)]
+pub(crate) struct TimeTriggerDeserializer;
+
+#[cfg(feature = "config_parsing")]
+impl Deserialize for TimeTriggerDeserializer {
+    type Trait = dyn Trigger;
+
+    type Config = TimeTriggerConfig;
+
+    fn deserialize(
+        &self,
+        config: TimeTriggerConfig,
+        _: &Deserializers,
+    ) -> anyhow::Result<Box<dyn Trigger>> {
+        Ok(Box::new(TimeTrigger::new(config.limit)))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use mock_instant::MockClock;
+    use std::time::Duration;
+
+    fn trigger_with_time(limit: TimeTriggerLimit, millis: u64) -> (bool, bool) {
+        let file = tempfile::tempdir().unwrap();
+        let logfile = LogFile {
+            writer: &mut None,
+            path: file.path(),
+            len: 0,
+        };
+
+        let trigger = TimeTrigger::new(limit);
+
+        MockClock::advance_system_time(Duration::from_millis(millis / 2));
+        let result1 = trigger.trigger(&logfile).unwrap();
+
+        MockClock::advance_system_time(Duration::from_millis(millis / 2));
+        let result2 = trigger.trigger(&logfile).unwrap();
+
+        (result1, result2)
+    }
+
+    #[test]
+    fn trigger() {
+        // Second
+        let second_in_milli = 1000;
+        let minute_in_milli = second_in_milli * 60;
+        let hour_in_milli = minute_in_milli * 60;
+        let day_in_milli = hour_in_milli * 24;
+        let week_in_milli = day_in_milli * 7;
+        let month_in_milli = day_in_milli * 31;
+        let year_in_milli = day_in_milli * 366;
```

Comment:
Why 366 here? I tried searching the docs for any note of it. I assume that is covering the leap year but want to verify

Reply:
Yeah, it's covering the leap year. And I think it's unnecessary now because both 1970 and 1971 are not leap year. Thanks.